src/libmw/include/mw/models/tx/UTXO.h

```diff
@@ -34,4 +37,45 @@ class UTXO : public Traits::ISerializable
     int32_t m_blockHeight;
     mmr::LeafIndex m_leafIdx;
     Output m_output;
+};
+
+/// <summary>
+/// MWEB UTXO wrapper that supports serialization into multiple formats.
+/// </summary>
+class NetUTXO
+{
+public:
+    static const uint8_t FULL_UTXO = 0x00;
+    static const uint8_t HASH_ONLY = 0x01;
+    static const uint8_t COMPACT_UTXO = 0x02;
+
+    NetUTXO() = default;
+    NetUTXO(const uint8_t format, const UTXO::CPtr& utxo)
+        : m_format(format), m_utxo(utxo) { }
+
+    template <typename Stream>
+    inline void Serialize(Stream& s) const
+    {
+        s << m_utxo->GetBlockHeight();
```

Comment:
Why is block height needed here?

Reply:
So that we can assign a timestamp to the tx record for the utxo. Also confirmed utxos require knowledge of a confirmation block.

Reply:
You already have that info from the MWEB headers that you downloaded (output MMR size)

Reply:
We don't download all the headers, only the last one.

Reply:
Oh shoot, you're right. Well, this could be an issue. It's not really safe to include block height here, since we have no way of verifying it. Every other field that's returned in the LIP-0006 messages (https://github.com/DavidBurkett/lips/blob/LIP0006/LIP-0006.mediawiki) is verifiable, but I don't think there's a straightforward way to do that for block_height privately and efficiently.